azurerm/resource_arm_sql_virtual_network_rule.go

```diff
@@ -0,0 +1,141 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/sql/mgmt/2015-05-01-preview/sql"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSqlVirtualNetworkRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSqlVirtualNetworkRuleCreateUpdate,
+		Read:   resourceArmSqlVirtualNetworkRuleRead,
+		Update: resourceArmSqlVirtualNetworkRuleCreateUpdate,
+		Delete: resourceArmSqlVirtualNetworkRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				//TODO: Validation (invalid if non-numeric characters)
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"server_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"virtual_network_subnet_id": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
should this be `ForceNew` or is it possible to migrate between Virtual Networks? Can we add a test for this if that's the case

Reply:
It does indeed look like you can change the subnets of a rule, and it does apply correctly. I will make a task to create a test case for that. 😄 

On a side note, I also tested migrating from one SQL server to another without `ForceNew` (just in case) , and it creates a rule on the 2nd target SQL server, but does not delete it on the first server, so the `ForceNew` attribute is needed on the `server_name`.

Reply:
✅